api/filter/accesslog.proto

```diff
@@ -285,8 +319,22 @@ message StreamAccessLogsMessage {
   // structured metadata and is a performance optimization.
   Identifier identifier = 1;
 
-  // A list of access logs.
-  repeated AccessLogEntry logs = 2;
+  // Wrapper for batches of HTTP access log entries.
+  message HTTPAccessLogEntries {
+    repeated HTTPAccessLogEntry log_entry = 1;
+  }
+
+  // Wrapper for batches of TCP access log entries.
+  message TCPAccessLogEntries {
+    repeated TCPAccessLogEntry log_entry = 1;
+  }
+
+  // Batches of log entries of a single type. Generally speaking, a given stream should only
+  // ever incude one type of log entry.
+  oneof log_entries {
+    HTTPAccessLogEntries http_logs = 2;
```

Comment:
`request local storage` also needs to be logged here.  For example If `jwt.principal` is extracted, it would be good to make it as part of the access log record.

`request local storage` can map to Mixer attributes.  
@spikecurtis

Reply:
What does "request local storage" mean?

Reply:
It is the mechanism by which filters propagate information without resorting to storing it in headers. Is there a specific name for that concept in envoy?
https://github.com/envoyproxy/data-plane-api/pull/222#issuecomment-342658567
You used it here before so I am just sticking to that nomenclature.

Reply:
We don't actually have this yet. I think the idea is that we will use opaque metadata for this purpose (just a proto Struct). This is what we are doing in the access log.

Reply:
I see it here `google.protobuf.Struct metadata = 25;` 
Is there a proposal for this? 
It would be great to have these data items individually addressable.
Many are plain strings, like `jwt.principal` so a few types cover a lot.

Equivalent of `map<string, Value>` with
```message Value {
oneof {
   string
   int 
   google.protobuf.Struct  
}
}```

it makes apache style `%{VARNAME}n` filters possible

Reply:
Just so I'm clear, is the request here to modify metadata to support nested maps?

Reply:
Let's skip this for now. The basic metadata is fine.